enhancements/network/user-defined-network-segmentation.md

```diff
@@ -391,6 +488,98 @@ guarantee uniqueness and eliminates the ability to falsify access.
+
+The underlying OVN network name, specified by the CNI conf `name` field, should be unique.
+Having the network name unique and non-configurable, mitigates the risk where a malicious entity with access to namespace "A" workloads could tap into network on namespace "B" by guessing its name.
+
+Creating namespace scoped CRD instance should trigger creation of a corresponding NAD at the namespace the CRD instance reside.
+Following the above [example](#example---namespace-scoped-network) above, the following NAD should be created:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: db-network
+  namespace: demo
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: OVNNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"8301b3c4-ff07-4c6d-9786-232e02b87794",
+          "netAttachDefName":"demo/poc-db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "primaryNetwork": "true"
+      }'
+```
+
+Creating cluster scoped CRD instance should trigger creation of the corresponding NAD at each namespace specified by `spec.selector`.
+Following the above [example](#example---cluster-scoped-network) above, the following NADs should be created:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: db-network
+  namespace: mynamespace 
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: ClusterOVNNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"ef1df435-ef21-4850-a4a2-5fd8a8a8f991",
+          "netAttachDefName":"mynamespace/db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "primaryNetwork": true
+      }'
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: db-network
+  namespace: theirnamespace
+  ownerReferences:
+  - apiVersion: k8s.ovn.org/v1alpha1
+    blockOwnerDeletion: true
+    kind: ClusterOVNNetwork
+    name: db-network
+    uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >
+    '{
+        "cniVersion":"0.3.1",
+        "excludeSubnets":"10.0.0.100/24",
+        "mtu":1500,   
+        "name":"ef1df435-ef21-4850-a4a2-5fd8a8a8f991",   <--- same name as in other namespaces
+        "netAttachDefName":"theirnamespace/db-network",
+        "subnets":"10.0.0.0/24",
+        "topology":"layer2",
+        "type":"ovn-k8s-cni-overlay",
+        "primaryNetwork": true
+    }'
+```
```

Comment:
Please add a validations section for the CRD approach that outlines some scenarios you plan to address for a better user experience:

1. what if admin has created NAD for nsA and user has created CR for same namespace in a different way? what would you do for overlapping configurations?
2. what if pods already exist in a namespace attached to the default network?
3. what if namespace scoped NAD is requesting a different network for a given namespace while cluster-scoped nad is requesting a different network for that same namespace? who wins?
4. what if multiple cluster-scoped NADs are selecting same namespaces where both networks are primary?